### ISSUE TITLE
Do you have the pytorch version of this part? I don't quite understand the author's approachBy the way, zongze, can you share your pretrained face segmentation model? @betterze

### ISSUE THREAD

**Issue by jweihe** [betterze/StyleSpace#28]
By the way, zongze, can you share your pretrained face segmentation model? @betterze 
And, how about the speed of gradient computation (I see that you use 1K images to compute gradient)?
I find in my experiments, the speed of only an image using your code is very slow... (maybe 1 image need 40 minutes), thus do you have any ways to speed it up?

_Originally posted by @sunpeng1996 in https://github.com/betterze/StyleSpace/issues/4#issuecomment-826317931_

**Comment by jweihe** [comment]
Do you have the pytorch version of this part? I don't quite understand the author's approach

**Comment by betterze** [comment]
The pretrained face segmentation model is [here](https://github.com/zllrunning/face-parsing.PyTorch). I attatch the link in the [Localized channels section](https://github.com/betterze/StyleSpace#localized-channels). 

The gradient computation takes time. It also takes me around 40 mins per image since the stylegan2 in 1024x1024 is very big. I use many gpu to compute the gradient. Each gpu will be assigned with 4 images. You can change this parameters.

I have torch version of codes for single channel manipulation in S space. But I dont have gradient computation codes for torch version.

**Comment by jweihe** [comment]
thank you，I have implemented the code myself~

**Comment by lovejing0306** [comment]
> thank you，I have implemented the code myself~

Can you share your pytorch code ?

**Comment by WuYufeng98** [comment]
> thank you，I have implemented the code myself~
Can you share your pytorch code ?

**Comment by betterze** [comment]
We implement the single channel manipulation in torch, pls take a look of [here](https://github.com/orpatashnik/StyleCLIP/blob/main/global_torch/manipulate.py).

**Comment by WuYufeng98** [comment]
> We implement the single channel manipulation in torch, pls take a look of [here](https://github.com/orpatashnik/StyleCLIP/blob/main/global_torch/manipulate.py).
thank you！
